pkg/provisioner/worker_test.go

```diff
@@ -0,0 +1,278 @@
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				UID:       "second-uid-420",
+				Name:      "second-name",
+				Namespace: "first-namespace",
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				UID:       "third-uid-420",
+				Name:      "third-name",
+				Namespace: "second-namespace",
+			},
+		},
+	}
+
+	wantWorkers := []Worker{
+		{
+			ID:     "first-uid-420",
+			Name:   fmt.Sprintf("%s/%s", "first-namespace", "first-name"),
+			Status: worker.StatusUnknown,
+		},
+		{
+			ID:     "second-uid-420",
+			Name:   fmt.Sprintf("%s/%s", "first-namespace", "second-name"),
+			Status: worker.StatusUnknown,
+		},
+		{
+			ID:     "third-uid-420",
+			Name:   fmt.Sprintf("%s/%s", "second-namespace", "third-name"),
+			Status: worker.StatusUnknown,
+		},
+	}
+	gotWorkers := convertPodsToWorkers(pods)
+
+	assert.ElementsMatch(t, wantWorkers, gotWorkers)
+}
+
+func TestConvertPodListToWorkerList_NilPodListReturnsEmptyWorkerList(t *testing.T) {
+	wantWorkerList := WorkerList{}
+	gotWorkerList := convertPodListToWorkerList(nil)
+	assert.ElementsMatch(t, wantWorkerList.Items, gotWorkerList.Items)
+	assert.Equal(t, wantWorkerList.Count, gotWorkerList.Count)
+}
+
+func TestConvertPodListToWorkerList_Success(t *testing.T) {
+	podList := v1.PodList{
+		Items: []v1.Pod{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					UID:       "first-uid-420",
+					Name:      "first-name",
+					Namespace: "first-namespace",
+				},
+			},
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					UID:       "second-uid-420",
+					Name:      "second-name",
+					Namespace: "first-namespace",
+				},
+			},
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					UID:       "third-uid-420",
+					Name:      "third-name",
+					Namespace: "second-namespace",
+				},
+			},
+		},
+	}
+
+	wantWorkerList := WorkerList{
+		Items: []Worker{
+			{
+				ID:     "first-uid-420",
+				Name:   fmt.Sprintf("%s/%s", "first-namespace", "first-name"),
+				Status: worker.StatusUnknown,
+			},
+			{
+				ID:     "second-uid-420",
+				Name:   fmt.Sprintf("%s/%s", "first-namespace", "second-name"),
+				Status: worker.StatusUnknown,
+			},
+			{
+				ID:     "third-uid-420",
+				Name:   fmt.Sprintf("%s/%s", "second-namespace", "third-name"),
+				Status: worker.StatusUnknown,
+			},
+		},
+	}
+	wantWorkerList.Count = len(wantWorkerList.Items)
+
+	gotWorkerList := convertPodListToWorkerList(&podList)
+
+	assert.ElementsMatch(t, wantWorkerList.Items, gotWorkerList.Items)
+	assert.Equal(t, wantWorkerList.Count, gotWorkerList.Count)
+}
+
+func makeTestPod(initContainerStatuses []v1.ContainerStatus, containerStatuses []v1.ContainerStatus, conditions []v1.PodCondition) v1.Pod {
+	return v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			UID:       "some-uid-420",
+			Name:      "some-name",
+			Namespace: "some-namespace",
+		},
+		Status: v1.PodStatus{
+			Conditions:            conditions,
+			InitContainerStatuses: initContainerStatuses,
+			ContainerStatuses:     containerStatuses,
+		},
+	}
+}
```

Comment:
Uses of this function was difficult to read IMO. Your invokations had nil in either the first, second, third argument, or combinations thereof.

Suggest to instead just pass in the `v1.PodStatus` and let the callers declare verbosely if using the `Conditions`, `InitContainerStatuses`, or `ContainerStatuses` fields.